### PR TITLE
BE: Add automatic username generation

### PR DESCRIPTION
**Pull request by Jozys**
- Usernames of Ory Users are now automatically generated and created within the user in the database
- This replaces the uuid usernames, that were used before that

### CODE REVIEW

File: backend/src/auth/auth.service.ts [hunk 1]
```diff
@@ -66,4 +71,17 @@ export class AuthService {
...
+
+  /**
+   * Generates an username, repeated aslong as the generated username already exists
+   * @returns username a random string of up to 20 characters
+   * @private this method is only used in the auth service
+   */
+  private async generateUsername(): Promise<string> {
+    let username = generateUsername('', 0, 15);
+    while (await this.userService.checkUsernameExists(username)) {
+      username = generateUsername('', 0, 20);
```
Comment (Integraluminium) [inline backend/src/auth/auth.service.ts:83]: Why are you using different lengths?

File: backend/src/auth/auth.service.ts [hunk 2]
```diff
@@ -66,4 +71,17 @@ export class AuthService {
...
     }
   }
+
+  /**
+   * Generates an username, repeated aslong as the generated username already exists
+   * @returns username a random string of up to 20 characters
+   * @private this method is only used in the auth service
+   */
+  private async generateUsername(): Promise<string> {
+    let username = generateUsername('', 0, 15);
```
Comment (Integraluminium) [inline backend/src/auth/auth.service.ts:81]: Maybe read the values from length from the config file

**Review by Integraluminium — CHANGES_REQUESTED** [review]
Looks good, but there are some minor questions.

File: backend/src/auth/auth.service.ts (hunk 1 above)
Comment (Jozys) [reply backend/src/auth/auth.service.ts:83]: That was a copy/paste mistake from my side.

**Comment by Jozys** [comment]
I resolved your conversations. Please review my changes :)

**Review by Integraluminium — APPROVED** [review]
OK

**Pull request merged**